src/controllers/auth.ts

```diff
@@ -0,0 +1,108 @@
+import type * as express from 'express'
+
+import { randomBytes } from 'node:crypto'
+import { Get, Produces, Query, Request, Route, SuccessResponse } from 'tsoa'
+import { inject, injectable, singleton } from 'tsyringe'
+
+import { Env } from '../env.js'
+import { ForbiddenError, InternalError } from '../errors.js'
+import { Logger, type ILogger } from '../logger.js'
+import IDPService from '../models/idpService.js'
+import { HTMLController } from './HTMLController.js'
+
+function base64URLEncode(buf: Buffer) {
+  return buf.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/=/g, '')
+}
+
+const nonceCookieOpts: express.CookieOptions = {
+  sameSite: true,
+  httpOnly: true,
+  signed: true,
+  secure: true,
+  path: '/auth/redirect',
+}
+
+const tokenCookieOpts: express.CookieOptions = {
+  sameSite: true,
+  maxAge: 24 * 60 * 60 * 1000,
+  httpOnly: true,
```

Comment:
http? why not https? due to (kind/local)?

Reply:
I think you're misunderstanding what this does https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#httponly